todo/repositories/team_repository.py

```
@@ -0,0 +1,79 @@
+    @classmethod
+    def get_by_id(cls, team_id: str) -> Optional[TeamModel]:
+        """
+        Get a team by its ID.
+        """
+        teams_collection = cls.get_collection()
+        try:
+            team_data = teams_collection.find_one({"_id": ObjectId(team_id), "is_deleted": False})
+            if team_data:
+                return TeamModel(**team_data)
+            return None
+        except Exception:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)